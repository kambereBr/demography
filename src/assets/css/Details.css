.details-header {
  text-align: right;
  background-color: var(--bg-color);
  color: var(--text-color);
}

.demography-details {
  height: 25px;
  text-align: left;
  background-color: var(--primary-color);
  font-weight: 600;
  padding-left: 20px;
  padding-top: 3px;
}

.demography-list {
  padding-left: 0;
  margin-top: 0;
}

.demography-list > .demography-card:nth-child(odd) {
  background-color: var(--bg-color);
}

.details-header-info {
  display: flex;
  flex-direction: column;
  height: 150px;
  justify-content: center;
  margin-right: 15px;
}

.country-code,
.country-name {
  font-size: 1.5rem;
  font-weight: 600;
}
