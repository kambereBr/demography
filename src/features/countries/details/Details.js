import { React, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router';
import { v4 as uuidv4 } from 'uuid';
import { getCountryByIso3, fetchCountries } from '../countriesSlice';
import Demography from '../../demography/Demography';
import '../../../assets/css/Details.css';

const Details = () => {
  const { iso3 } = useParams();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCountries());
  }, [dispatch]);
  const details = useSelector((state) => getCountryByIso3(state, iso3));
  const recentDemography = details ? details.demography[details.demography.length - 1].value : 0;

  return (
    <div className="details-container">
      <div className="details-header">
        <div className="details-header-info">
          <span className="country-code">{details?.code}</span>
          <span className="country-name">{details?.country}</span>
          <span className="country-population">{`${recentDemography} people`}</span>
        </div>
        <div className="demography-details">
          <span>DEMOGRAPHY DETAILS</span>
        </div>
      </div>
      <div>
        <ul className="demography-list">
          {details && (details.demography.map((demography) => (
            <Demography
              key={uuidv4()}
              year={demography.year}
              populationCounts={demography.value}
            />
          )))}
        </ul>
      </div>
    </div>
  );
};

export default Details;
