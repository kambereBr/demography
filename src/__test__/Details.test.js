import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Details from '../components/countries/Details';

describe('Render Details correctly', () => {
  it('Render Details Screen correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Details />
        </MemoryRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
