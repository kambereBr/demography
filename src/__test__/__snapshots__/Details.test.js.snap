// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Details correctly Render Details Screen correctly 1`] = `
<div
  className="details-container"
>
  <div
    className="details-header"
  >
    <div
      className="details-header-info"
    >
      <span
        className="country-code"
      />
      <span
        className="country-name"
      />
      <span
        className="country-population"
      >
        0 people
      </span>
    </div>
    <div
      className="demography-details"
    >
      <span>
        DEMOGRAPHY DETAILS
      </span>
    </div>
  </div>
  <div>
    <ul
      className="demography-list"
    />
  </div>
</div>
`;
